.github/workflows/build.yml

```diff
@@ -105,44 +145,43 @@ jobs:
         shell: bash
 
         # keep versions consistent with the first and last from the strategy matrix
-        if: ${{ contains(fromJSON('["3.9"]'), matrix.python) }}
+        if: ${{ contains(fromJSON('["3.9"]'), matrix.python) && steps.secret-check.outputs.secrets_available == 'true'}}
         run: |
-          if [ -z "${{ secrets.encrypted_d17283647768_key }}" ]  || [ -z "${{ secrets.encrypted_d17283647768_key }}" ]; then
-            echo "No test configuration decryption keys available, skipping integration tests"
+          # decrypt the encrypted test synapse configuration
```

Comment:
This all shows as changed because I removed the `if` check here - since I am doing it earlier in the steps.

The only notable change from this block is L175:
```
          if [ ${{ steps.aws-check.outputs.run_opentelemetry }} == "true" ]; then
            export SYNAPSE_OTEL_INTEGRATION_TEST_EXPORTER="otlp"
          fi
```